main_test.go

```diff
@@ -230,6 +238,83 @@ var _ = Describe("Main", func() {
 			})
 		})
 	})
+
+	Describe("metricProcessor", func() {
+		type tokenJSON struct {
```

Comment:
TBH, checking the code https://github.com/alphagov/paas-cf-apps-statsd/blob/8a7fd4a2e54b0354b0d8ba7bf95c38a6e58e8385/vendor/golang.org/x/oauth2/internal/token.go#L208-L220 maybe if we return a `Content-Type` of `application/x-www-form-urlencoded`, `text/plain` we could use the `oauth.Token`